javalib/src/main/scala/java/net/URI.scala

```diff
@@ -1,33 +1,1415 @@
+    if (scheme == null && uri.scheme != null) {
+      -1
+    } else if (scheme != null && uri.scheme == null) {
+      1
+    } else if (scheme != null && uri.scheme != null) {
+      ret = scheme.compareToIgnoreCase(uri.scheme)
+      if (ret != 0) {
+        return ret
+      }
+    }
+    if (!opaque && uri.opaque) {
+      -1
+    } else if (opaque && !uri.opaque) {
+      1
+    } else if (opaque && uri.opaque) {
+      ret = schemespecificpart.compareTo(uri.schemespecificpart)
+      if (ret != 0) {
+        return ret
+      }
+    } else {
+      if (authority != null && uri.authority == null) {
+        1
+      } else if (authority == null && uri.authority != null) {
+        -1
+      } else if (authority != null && uri.authority != null) {
+        if (host != null && uri.host != null) {
+          if (userinfo != null && uri.userinfo == null) {
+            1
+          } else if (userinfo == null && uri.userinfo != null) {
+            -1
+          } else if (userinfo != null && uri.userinfo != null) {
+            ret = userinfo.compareTo(uri.userinfo)
+            if (ret != 0) {
+              return ret
+            }
+          }
+          ret = host.compareToIgnoreCase(uri.host)
+          if (ret != 0) {
+            return ret
+          }
+          if (port != uri.port) {
+            port - uri.port
+          }
+        } else {
+          ret = authority.compareTo(uri.authority)
+          if (ret != 0) {
+            return ret
+          }
+        }
+      }
+      ret = path.compareTo(uri.path)
+      if (ret != 0) {
+        return ret
+      }
+      if (query != null && uri.query == null) {
+        1
+      } else if (query == null && uri.query != null) {
+        -1
+      } else if (query != null && uri.query != null) {
+        ret = query.compareTo(uri.query)
+        if (ret != 0) {
+          return ret
+        }
+      }
+    }
+    if (fragment != null && uri.fragment == null) {
+      1
+    } else if (fragment == null && uri.fragment != null) {
+      -1
+    } else if (fragment != null && uri.fragment != null) {
+      ret = fragment.compareTo(uri.fragment)
+      if (ret != 0) {
+        return ret
+      }
+    }
+    0
+  }
+
+  private def duplicate(): URI = {
+    val clone: URI = new URI()
+    clone.absolute = absolute
+    clone.authority = authority
+    clone.fragment = fragment
+    clone.host = host
+    clone.opaque = opaque
+    clone.path = path
+    clone.port = port
+    clone.query = query
+    clone.scheme = scheme
+    clone.schemespecificpart = schemespecificpart
+    clone.userinfo = userinfo
+    clone.serverAuthority = serverAuthority
+    clone
+  }
+
+  /*
+   * Takes a string that may contain hex sequences like %F1 or %2b and
+   * converts the hex values following the '%' to lowercase
+   */
+
+  private def convertHexToLowerCase(s: String): String = {
+    val result: StringBuilder = new StringBuilder("")
+    if (s.indexOf('%') == -1) {
+      s
```

Comment:
You probably want a `return` in here. Otherwise the value of `s` is just going to be ignored, and execution will continue.